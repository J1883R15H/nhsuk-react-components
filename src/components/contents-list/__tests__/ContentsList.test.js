import React from 'react';
import { mount } from 'enzyme';
import ContentsList from '../contents-list';

describe('ContentsList', () => {
  it('item matches snapshot', () => {
    const mountedComponent = mount(
      <ContentsList.Item>Test Text</ContentsList.Item>
    );
    expect(mountedComponent).toMatchSnapshot();
    mountedComponent.unmount();
  });
  it('current item matches snapshot', () => {
    const mountedComponent = mount(
      <ContentsList.Item current>Test Text</ContentsList.Item>
    );
    expect(mountedComponent).toMatchSnapshot();
    mountedComponent.unmount();
  });
  it('container matches snapshot', () => {
    const mountedComponent = mount(
      <ContentsList>
        <ContentsList.Item>Test</ContentsList.Item>
        <ContentsList.Item current>Test Text</ContentsList.Item>
      </ContentsList>
    );

    expect(mountedComponent).toMatchSnapshot();
    mountedComponent.unmount();
  });
  it('responds to onClick', () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <ContentsList>
        <ContentsList.Item onClick={onClick}>Test</ContentsList.Item>
      </ContentsList>
    );
    const wrapperForAnonFunc = mount(
      <ContentsList>
        <ContentsList.Item>Test</ContentsList.Item>
      </ContentsList>
    );
    wrapper.find('a').simulate('click');
    wrapperForAnonFunc.find('a').simulate('click');
    expect(onClick).toHaveBeenCalled();
    wrapper.unmount();
  });
});
