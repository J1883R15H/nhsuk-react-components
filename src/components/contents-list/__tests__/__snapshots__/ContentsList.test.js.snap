// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentsList container matches snapshot 1`] = `
<ContentsList
  className=""
  role="navigation"
  visuallyHiddenText="Contents"
>
  <nav
    className="nhsuk-contents-list"
    role="navigation"
  >
    <h2
      className="nhsuk-u-visually-hidden"
    >
      Contents
    </h2>
    <ol
      className="nhsuk-contents-list__list"
    >
      <Item
        className=""
        current={false}
        href="#"
        id=""
        onClick={[Function]}
      >
        <li
          className="nhsuk-contents-list__item"
          id=""
        >
          <a
            className="nhsuk-contents-list__link"
            href="#"
            id="null"
            onClick={[Function]}
          >
            Test
          </a>
        </li>
      </Item>
      <Item
        className=""
        current={true}
        href="#"
        id=""
        onClick={[Function]}
      >
        <li
          className="nhsuk-contents-list__item"
          id=""
        >
          <span
            className="nhsuk-contents-list__current"
          >
            Test Text
          </span>
        </li>
      </Item>
    </ol>
  </nav>
</ContentsList>
`;

exports[`ContentsList current item matches snapshot 1`] = `
<Item
  className=""
  current={true}
  href="#"
  id=""
  onClick={[Function]}
>
  <li
    className="nhsuk-contents-list__item"
    id=""
  >
    <span
      className="nhsuk-contents-list__current"
    >
      Test Text
    </span>
  </li>
</Item>
`;

exports[`ContentsList item matches snapshot 1`] = `
<Item
  className=""
  current={false}
  href="#"
  id=""
  onClick={[Function]}
>
  <li
    className="nhsuk-contents-list__item"
    id=""
  >
    <a
      className="nhsuk-contents-list__link"
      href="#"
      id="null"
      onClick={[Function]}
    >
      Test Text
    </a>
  </li>
</Item>
`;
